agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/ConnectionString.java

```diff
@@ -87,7 +87,7 @@ private static void mapToConnectionConfiguration(
     // get ikey
     String instrumentationKey = kvps.get(Keywords.INSTRUMENTATION_KEY);
     if (Strings.isNullOrEmpty(instrumentationKey)) {
-      throw new InvalidConnectionStringException("Missing '" + Keywords.INSTRUMENTATION_KEY + "'");
+      logger.warn("New connection string is null or empty");
```

Comment:
I didn't follow why this change?

Reply:
we don't want to throw an exception when instrumentation key is empty.  null or empty is accepted.

Reply:
maybe in `parseInto` above we can have a separate path empty connection string, that way we don't lose this validation when connection string is not empty, and hopefully we don't need to `warn` in an expected case

separately, i noticed in this method we probably want to reset all the urls in `EndpointProvider` to the defaults before we start populating them, otherwise we could pick up some values left over from the prior connection string